src/view/region.rs

```diff
@@ -0,0 +1,410 @@
+    where
+        Self: 'a;
+
+    type NodePortOffsets<'a> = <RegionInternalGraph<'g, G> as PortView>::NodePortOffsets<'a>
+    where
+        Self: 'a;
+
+    delegate! {
+        to self.graph {
+            fn port_direction(&self, port: impl Into<PortIndex>) -> Option<Direction>;
+            fn port_node(&self, port: impl Into<PortIndex>) -> Option<NodeIndex>;
+            fn port_offset(&self, port: impl Into<PortIndex>) -> Option<crate::PortOffset>;
+            fn port_index(&self, node: NodeIndex, offset: crate::PortOffset) -> Option<PortIndex>;
+            fn ports(&self, node: NodeIndex, direction: Direction) -> Self::NodePorts<'_>;
+            fn all_ports(&self, node: NodeIndex) -> Self::NodePorts<'_>;
+            fn input(&self, node: NodeIndex, offset: usize) -> Option<PortIndex>;
+            fn output(&self, node: NodeIndex, offset: usize) -> Option<PortIndex>;
+            fn num_ports(&self, node: NodeIndex, direction: Direction) -> usize;
+            fn port_offsets(&self, node: NodeIndex, direction: Direction) -> Self::NodePortOffsets<'_>;
+            fn all_port_offsets(&self, node: NodeIndex) -> Self::NodePortOffsets<'_>;
+            fn contains_node(&self, node: NodeIndex) -> bool;
+            fn contains_port(&self, port: PortIndex) -> bool;
+            fn is_empty(&self) -> bool;
+            fn node_count(&self) -> usize;
+            fn port_count(&self) -> usize;
+            fn nodes_iter(&self) -> Self::Nodes<'_>;
+            fn ports_iter(&self) -> Self::Ports<'_>;
+            fn node_capacity(&self) -> usize;
+            fn port_capacity(&self) -> usize;
+            fn node_port_capacity(&self, node: NodeIndex) -> usize;
+        }
+    }
+}
+
+impl<'g, G> LinkView for Region<'g, G>
+where
+    G: LinkView,
+{
+    type LinkEndpoint = G::LinkEndpoint;
+
+    type Neighbours<'a> = <RegionInternalGraph<'g, G> as LinkView>::Neighbours<'a>
+    where
+        Self: 'a;
+
+    type NodeConnections<'a> = <RegionInternalGraph<'g, G> as LinkView>::NodeConnections<'a>
+    where
+        Self: 'a;
+
+    type NodeLinks<'a> = <RegionInternalGraph<'g, G> as LinkView>::NodeLinks<'a>
+    where
+        Self: 'a;
+
+    type PortLinks<'a> = <RegionInternalGraph<'g, G> as LinkView>::PortLinks<'a>
+    where
+        Self: 'a;
+
+    delegate! {
+        to self.graph {
+            fn get_connections(&self, from: NodeIndex, to: NodeIndex) -> Self::NodeConnections<'_>;
+            fn port_links(&self, port: PortIndex) -> Self::PortLinks<'_>;
+            fn links(&self, node: NodeIndex, direction: Direction) -> Self::NodeLinks<'_>;
+            fn all_links(&self, node: NodeIndex) -> Self::NodeLinks<'_>;
+            fn neighbours(&self, node: NodeIndex, direction: Direction) -> Self::Neighbours<'_>;
+            fn all_neighbours(&self, node: NodeIndex) -> Self::Neighbours<'_>;
+            fn link_count(&self) -> usize;
+        }
+    }
+}
+
+impl<'g, G> MultiView for Region<'g, G>
```

Comment:
It could very well be just a type alias.
I don't like that it's easy to expose the internal type in docs, which may be confusing to the user.

But I guess it's not worth the amount of extra code.